src/main/java/uk/ac/ebi/ega/accession/file/rest/HashValidator.java

```diff
@@ -34,16 +34,17 @@ public boolean isValid(FileModel fileModel, ConstraintValidatorContext constrain
 
         constraintValidatorContext.disableDefaultConstraintViolation();
 
-        if (fileModel.getHashType() == null) {
-            constraintValidatorContext.buildConstraintViolationWithTemplate(
-                    "Please provide a hashType.Supported types :MD5,SHA1").addConstraintViolation();
+        if (fileModel.getFileMd5() == null || fileModel.getFileMd5().length() != 32 ||
+            !fileModel.getFileMd5().equals(fileModel.getFileMd5().toLowerCase())) {
```

Comment:
Move this check of lower case to a private method boolean isLowerCase(String) to improve readability. That would have helped to detect NPE in the following case. Another thing to avoid that error is to use Objects.equals in a regular basis.